development/progress-bar-app/components/progress-bar/UploadDownloadProgress/UploadDownloadProgress.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useRef, useEffect } from "react";
```

Comment:
You can separate this in different examples. Having three examples with different behaviour in a single file is not appropriate. Also it will make it harder to directly copy code and may cause confusion. If there are any examples with multiple different components with different implementation, pls separate them in different examples.